core/src/main/kotlin/plugability/DokkaPlugin.kt

```diff
@@ -9,13 +9,32 @@ import kotlin.properties.ReadOnlyProperty
 import kotlin.reflect.KProperty
 import kotlin.reflect.KProperty1
 
+@RequiresOptIn(
+    level = RequiresOptIn.Level.WARNING,
+    message = "All of Dokka's plugin API is in preview and it can be changed " +
+            "in a backwards-incompatible manner with a best-effort migration. " +
+            "By opting in, you acknowledge the risks of relying on preview API."
+)
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.FIELD)
+@Retention(AnnotationRetention.BINARY)
+annotation class DokkaPluginApiPreview
+
+/**
+ * Acknowledgement for empty methods that inform users about [DokkaPluginApiPreview]
+ * Also, it allows to not propagates the annotation in IDE by default when a user autogenerate methods.
+ */
+@DokkaPluginApiPreview
+object PluginApiPreviewAcknowledgement
+
 abstract class DokkaPlugin {
     private val extensionDelegates = mutableListOf<KProperty<*>>()
     private val unsafePlugins = mutableListOf<Lazy<Extension<*, *, *>>>()
 
     @PublishedApi
     internal var context: DokkaContext? = null
 
+    @DokkaPluginApiPreview
+    protected abstract fun warnDokkaPluginApiIsInPreviewEmptyMethod(): PluginApiPreviewAcknowledgement
```

Comment:
It's unclear what the `EmptyMethod` in the name is supposed to indicate. The intended behaviour, I think, is that the user returns `PluginApiPreviewAcknowledgement`, so it will not be empty. I don't think it can considered to be empty even if it has the default `TODO()`. Proposing to rename it so that it's the same as the enum - `pluginApiPreviewAcknowledgement`.

Can you also add a `@see [PluginApiPreviewAcknowledgement]` to the KDoc (or vise versa)? Users will most likely first look at the documentation of the method itself, not the return type